src/Exporter/TargetStatsExporter.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Drupal\campaignion_csv\Exporter;
+
+use Drupal\campaignion_csv\Files\CsvFileInterface;
+use Drupal\campaignion_csv\Timeframe;
+
+/**
+ * Export message stats for email-to-target targets.
+ */
+class TargetStatsExporter {
+
+  /**
+   * The timeframe for the current export.
+   *
+   * @var \Drupal\campaignion_csv\Timeframe
+   */
+  protected $timeframe;
+
+  /**
+   * Create a new exporter based on the info.
+   */
+  public static function fromInfo(array $info) {
+    return new static($info['timeframe']);
+  }
+
+  /**
+   * Create a new instance.
+   *
+   * @param \Drupal\campaignion_csv\Timeframe $timeframe
+   *   Export activities within this timeframe.
+   */
+  public function __construct(Timeframe $timeframe) {
+    $this->timeframe = $timeframe;
+  }
+
+  /**
+   * Create a sorted list of targets and message counts.
+   */
+  protected function readTargetData() {
+    list($start, $end) = $this->timeframe->getTimestamps();
+    $sql = <<<SQL
+SELECT s.sid, d.data
+FROM webform_submissions s
+  INNER JOIN webform_component c USING(nid)
+  INNER JOIN webform_submitted_data d USING(nid, sid, cid)
+WHERE c.type='e2t_selector' AND s.submitted BETWEEN :start AND :end AND s.sid>:last_sid
+LIMIT 10000
+SQL;
+    $last_sid = 0;
+    $args = [':start' => $start, ':end' => $end - 1];
+    $message_counts = [];
+    $targets = [];
+    $continue = TRUE;
```

Comment:
Look like the variable is not used anywhere, maybe it was left-over?

Reply:
Fixed in the latest version.